corehq/motech/openmrs/atom_feed.py

```diff
@@ -93,7 +94,15 @@ def get_feed_xml(requests, feed_name, page):
               "longer be found.")
         )
         raise exception
-    root = etree.fromstring(resp.content)
+    try:
+        root = etree.fromstring(resp.content)
+    except etree.XMLSyntaxError as err:
+        requests.notify_exception(
+            str(err),
+            _('There is an XML syntax error in the OpenMRS Atom feed at '
+              f'"{resp.url}".')
+        )
+        raise OpenmrsFeedSyntaxError() from err
```

Comment:
I like accounting for potentially invalid XML, but is this the best approach to handle connection errors? In the scenario that was appearing on Sentry, the response was a 500, and the content was not XML. I don't know if security problems could arise from passing random text to an XML parser, but if the server is saying that there was an Internal Server Error, I don't really think we need to risk passing the response to the parser.

Reply:
It seems I got distracted by the XML parsing error, and totally forgot about the 500 error.

You're right @mjriley . The 500 error is the real issue. And it's caused by HQ requesting an invalid page. (The Atom feed is paginated.)

https://github.com/dimagi/commcare-hq/pull/31000 deals with the invalid page number.